src/dotnet/ForwardingApp.cs

```diff
@@ -1,28 +1,14 @@
-﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
-// Licensed under the MIT license. See LICENSE file in the project root for full license information.
-
+﻿using Microsoft.DotNet.Cli.Utils;
+using System;
 using System.Collections.Generic;
 using System.Diagnostics;
-using Microsoft.DotNet.Cli.Utils;
+using System.Text;
 
 namespace Microsoft.DotNet.Cli
 {
-    /// <summary>
-    /// A class which encapsulates logic needed to forward arguments from the current process to another process
-    /// invoked with the dotnet.exe host.
-    /// </summary>
     public class ForwardingApp
```

Comment:
The issue was with the test projects that reference the dotnet project.  Originally I just had `ForwardingApp` be internal in Cli.Utils, but that meant that things that derived from it in the dotnet project had to be internal, which meant that the test projects for dotnet couldn't use those types anymore.  Rather than adding InternalsVisibleTo from the dotnet project to all the test projects, I decided to split it up into internal and public pieces as you see now.